packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -277,6 +277,11 @@ export function getStylesDeclarations(
 		if ( typeof ruleValue !== 'string' && ruleValue?.ref ) {
 			const refPath = ruleValue.ref.split( '.' );
 			ruleValue = get( tree, refPath );
+			// Presence of another ref indicates a reference to another dynamic value.
+			// Pointing to another dynamic value is not supported.
+			if ( ! ruleValue || !! ruleValue?.ref ) {
```

Comment:
I ran out of time to [add tests](https://github.com/WordPress/gutenberg/blob/22fec1abe5bb61935f40db1aaed3e8b80d097d40/packages/edit-site/src/components/global-styles/test/use-global-styles-output.js) for this today. Will do later or in a follow up